YoutubeDLSharp/Utils.cs

```diff
@@ -80,5 +89,201 @@ public static string GetFullPath(string fileName)
+        public static string FfmpegBinaryName => GetFfmpegBinaryName();
+        public static string FfprobeBinaryName => GetFfprobeBinaryName();
+
+        public static async Task DownloadBinaries(bool skipExisting = true, string directoryPath = "")
+        {
+            if (skipExisting)
+            {
+                if(!File.Exists(Path.Combine(directoryPath, GetYtDlpBinaryName())))
+                {
+                    await DownloadYtDlp(directoryPath);
+                }
+                if (!File.Exists(Path.Combine(directoryPath, GetFfmpegBinaryName())))
+                {
+                    await DownloadFFmpeg(directoryPath);
+                }
+                if (!File.Exists(Path.Combine(directoryPath, GetFfprobeBinaryName())))
+                {
+                    await DownloadFFprobe(directoryPath);
+                }
+            }
+            else
+            {
+                await DownloadYtDlp(directoryPath);
+                await DownloadFFmpeg(directoryPath);
+                await DownloadFFprobe(directoryPath);
+            }            
+        }
+
+        private static string GetYtDlpBinaryName(bool fullPath = false)
+        {
+            const string BASE_GITHUB_URL = "https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp";
+
+            string downloadUrl;
+            switch (OSHelper.GetOSVersion())
+            {
+                case OSVersion.Windows:
+                    downloadUrl = $"{BASE_GITHUB_URL}.exe";
+                    break;
+                case OSVersion.OSX:
+                    downloadUrl = $"{BASE_GITHUB_URL}_macos";
+                    break;
+                case OSVersion.Linux:
+                    downloadUrl = BASE_GITHUB_URL;
+                    break;
+                default:
+                    throw new Exception("Your OS isn't supported");
+            }
+            return fullPath ? downloadUrl : Path.GetFileName(downloadUrl);
+        }
+
+        private static string GetFfmpegBinaryName()
+        {
+            switch (OSHelper.GetOSVersion())
+            {
+                case OSVersion.Windows:
+                    return "ffmpeg.exe";
+                case OSVersion.OSX:
+                case OSVersion.Linux:
+                    return "ffmpeg";
+                default:
+                    throw new Exception("Your OS isn't supported");
+            }            
+        }
+
+        private static string GetFfprobeBinaryName()
+        {
+            switch (OSHelper.GetOSVersion())
+            {
+                case OSVersion.Windows:
+                    return "ffprobe.exe";
+                case OSVersion.OSX:
+                case OSVersion.Linux:
+                    return "ffprobe";
+                default:
+                    throw new Exception("Your OS isn't supported");
+            }
+        }
+
+        internal static async Task DownloadYtDlp(string directoryPath = "")
```

Comment:
I think it might be useful to provide the option to download each binary separatly (as before). Maybe just make this method (and those for other binaries) public?